faiss/gpu/impl/IVFBase.cuh

```diff
@@ -212,6 +243,9 @@ class IVFBase {
     /// INDICES_CPU), then this maintains a CPU-side map of what
     /// (inverted list id, offset) maps to which user index
     std::vector<std::vector<Index::idx_t>> listOffsetToUserIndex_;
+
+    /// use this map to store the mapping of index to storage location
+    std::map<Index::idx_t, uint64_t> indexMap;
```

Comment:
change this to `unordered_map`, and can you make a struct for the index -> storage information instead of packing it into a `uint64_t` (something like `struct IVFLocation { uint32_t listId; uint32_t offset; };` or some such)